Adds Deploy to Connect Cloud buttons to app READMEs

garrettgman opened the pull request:
Related: https://github.com/posit-dev/py-shiny-templates/pull/25
Related: https://github.com/posit-dev/py-shinywidgets/pull/162

-----------------------

This PR adds a button to the README of each example app that allows users to quickly deploy the app to Connect Cloud and see it in action—with one click of the button. The PR does not otherwise affect the code in the apps or the repository. 

The buttons look like this.

<img width="1088" alt="Screenshot 2024-09-26 at 1 11 34 PM" src="https://github.com/user-attachments/assets/00b7fb0a-8363-4c4e-a91d-60e11a1955af">

When I tested it, every app deployed within seconds.

**Why add Deploy to Connect Cloud buttons?**

[Connect Cloud](https://connect.posit.cloud/) is part of Posit's strategy to provide value to Python users. With Connect Cloud, users can very quickly deploy and maintain Shiny apps straight from their Github repository. We believe in Connect Cloud as a company and want to highlight it alongside our own assets.

icarusz commented:
This looks good to me though should we put the button at the bottom, the description is not too long and it seems like you might want to do it _after_ reading the description rather than before.

garrettgman commented:
@schloerke I moved the buttons right as requested 👍

schloerke commented:
New look:

![Screenshot 2024-10-02 at 2 19 56 PM](https://github.com/user-attachments/assets/9faf6b82-05ac-4b46-acf5-959b6b4a8193)

Thank you! It will handle any description content as it is added.

schloerke commented:
A: Yes. It'll also require `requirements.txt` files for each app.

schloerke commented:
@garrettgman I added (untested) requirements.txt files and readme.md files for each examples root app dir. When you have a chance, please verify deployment works and looks as you'd expect.

garrettgman commented:
@schloerke Thank you for added those. I verified that deployment works for all of the buttons, excepting a few which I removed:
* Brownian app required us to include a large npm dependency in the repo to enable deployment, so I removed the deploy link
* chat app requires openai keys to deploy, so I removed the deploy buttons
* The hold app example in the Shiny express apps failed to deploy because within the app code, shiny.ui.output_code() called an unrecognized inline=TRUE argument, so I removed the argument. After that it deploys correctly.

schloerke commented on examples/express/hold_app.py:
```diff
@@ -25,4 +25,4 @@ def txt():
 
 ui.input_slider("n", "N", 1, 100, 50)
 
-shiny.ui.output_code("txt", inline=True)
```
Thank you! 

Reference: https://shiny.posit.co/py/api/core/ui.output_code.html

No `inline` argument

schloerke commented:
Thank you! Merging
